src/shared/install.c

```diff
@@ -2827,19 +2827,14 @@ static int presets_find_config(UnitFileScope scope, const char *root_dir, char *
         assert(scope >= 0);
         assert(scope < _UNIT_FILE_SCOPE_MAX);
 
-        switch (scope) {
-        case UNIT_FILE_SYSTEM:
+        if (scope == UNIT_FILE_SYSTEM)
                 dirs = (const char* const*) CONF_PATHS_STRV("systemd/system-preset");
-                break;
```

Comment:
is this really safe? I mean, it feels not much better, as it just uses the fact that for single-line if blocks you don't need `{}`... but that appears to syntactic trickery at most...

I figure this should just be a global static variable instead, that we assign here...

Reply:
If you prefer, I could rewrite this using a ternary operator, but I don't think it's much better.

I wouldn't want to create a global variable for this, because right now it's all nice and local. Local variables are used in all the other places where CONF_PATHS_STRV() is used.

Reply:
yeah, i think the ternary operator would be nicer. or  a static const variable or so...